src/Testcontainers.Qdrant/QdrantBuilder.cs

```diff
@@ -0,0 +1,109 @@
+using System.IO;
+using System.Security.Cryptography.X509Certificates;
+using Org.BouncyCastle.OpenSsl;
+using Org.BouncyCastle.Security;
+
+namespace Testcontainers.Qdrant;
+
+/// <inheritdoc cref="ContainerBuilder{TBuilderEntity, TContainerEntity, TConfigurationEntity}" />
+[PublicAPI]
+public sealed class QdrantBuilder : ContainerBuilder<QdrantBuilder, QdrantContainer, QdrantConfiguration>
+{
+	public const string QdrantImage = "qdrant/qdrant:v1.5.0";
+
+	public const ushort QdrantHttpPort = 6333;
+
+	public const ushort QdrantGrpcPort = 6334;
+
+	public const string QdrantLocalConfigurationFilePath = "/qdrant/config/local.yaml";
+	
+	public const string QdrantTlsCertFilePath = "/qdrant/tls/cert.pem";
+	
+	public const string QdrantTlsKeyFilePath = "/qdrant/tls/key.pem";
+
+	public QdrantBuilder() : this(new QdrantConfiguration()) =>
+		DockerResourceConfiguration = Init().DockerResourceConfiguration;
+
+	private QdrantBuilder(QdrantConfiguration dockerResourceConfiguration) : base(dockerResourceConfiguration) =>
+		DockerResourceConfiguration = dockerResourceConfiguration;
+
+	/// <summary>
+	/// A path to a configuration file with which configure the instance.
+	/// </summary>
+	/// <param name="configurationFilePath">The path to the configuration file</param>
+	public QdrantBuilder WithConfigFile(string configurationFilePath) =>
+		Merge(DockerResourceConfiguration, new QdrantConfiguration(configurationFilePath: configurationFilePath))
+			.WithBindMount(configurationFilePath, QdrantLocalConfigurationFilePath);
+    
+	/// <summary>
+	/// The API key used to secure the instance. A certificate should also be provided to <see cref="WithCertificate"/>
+	/// to enable TLS
+	/// </summary>
+	/// <param name="apiKey">The API key</param>
+	public QdrantBuilder WithApiKey(string apiKey) =>
+		Merge(DockerResourceConfiguration, new QdrantConfiguration(apiKey: apiKey))
+			.WithEnvironment("QDRANT__SERVICE__API_KEY", apiKey);
+	
+	/// <summary>
+	/// A certificate to use to enable Transport Layer Security (TLS). The certificate must contain the
+	/// private key.
+	/// </summary>
+	/// <param name="certificate">A certificate containing a private key</param>
+	public QdrantBuilder WithCertificate(X509Certificate2 certificate)
```

Comment:
Can we not expect the developer to provide valid `cert.pem` and `key.pem` files? This way, we can avoid dealing with all the certificate-related issues and Bouncy Castle.